### ISSUE TITLE
GetPassword limited to 254 characters on Windows

### ISSUE THREAD

**Issue by antgiant** [rclone/rclone#3798]
#### Output of `rclone version`
rclone v1.49.2
- os/arch: windows/amd64
- go version: go1.12.3

#### Describe the issue
At least on Windows, the RCLONE_CONFIG_PASS environment variable only uses the first 254 characters of the value. Both Windows environment variables and rclone configuration allow for much longer strings. However, setting the environment variable to a valid password that is more than 254 characters always results in a decryption error.

If this is a language limit then even just a note in the documentation would be appreciated.

**Comment by ncw** [comment]
I tried setting environment variables on Windows longer than 254 bytes and reading them with go - that worked just fine.

I think the limitation is when the password gets read in interactive configuration

I can replicate it with this small go program - you can't enter more than 254 characters.

```go
package main

import (
       "os"
       "fmt"
       "golang.org/x/crypto/ssh/terminal"
)

func main() {
        val, err := terminal.ReadPassword(int(os.Stdin.Fd()))
        fmt.Println(err)
        fmt.Printf("len=%d: %q\n", len(val), val)
}
```

I reported this as an upstream bug: https://github.com/golang/go/issues/36071

**Comment by antgiant** [comment]
Just to make sure we are saying the same thing. I can have a config password much longer than 254 characters and it works fine if I paste it into the interactive command prompt. However if I enter it into the RCLONE_CONFIG_PASS environment variable it does not work.

**Comment by ncw** [comment]
That is the conclusion I came to after a bit of messing around!

Hopefully this will get fixed upstream in the not too distant future and I can pull the fix in.  In the mean time I'll leave this issue open.

**Comment by mxdpeep** [comment]
what is the use case for 254+ bytes long password?

**Comment by ncw** [comment]
This has been fixed upstream now - I've pulled the fix into rclone.

Which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.51
